packages/query/src/block-processor.ts

```diff
@@ -0,0 +1,390 @@
+    const startHeight = lastBlockSynced ? lastBlockSynced + 1n : 0n;
+    const latestBlockHeight = await this.querier.tendermint.latestBlockHeight();
+
+    // this is an indefinite stream of the (compact) chain from the network
+    // intended to run continuously
+    for await (const compactBlock of this.querier.compactBlock.compactBlockRange({
+      startHeight,
+      keepAlive: true,
+      abortSignal: this.abortController.signal,
+    })) {
+      if (compactBlock.fmdParameters)
+        await this.indexedDb.saveFmdParams(compactBlock.fmdParameters);
+
+      if (compactBlock.gasPrices) await this.indexedDb.saveGasPrices(compactBlock.gasPrices);
+
+      // wasm view server scan
+      // - decrypts new notes
+      // - decrypts new swaps
+      // - updates idb with advice
+      //const shouldStoreProgress = await this.viewServer.scanBlock(compactBlock);
+      const wasmWantsFlush = await this.viewServer.scanBlock(compactBlock);
+
+      // flushing is slow, avoid it until
+      // - wasm says
+      // - every 1000th block
+      // - every block at tip
+      const flushReasons = {
+        wasmWantsFlush,
+        interval: compactBlock.height % 1000n === 0n,
+        new: compactBlock.height > latestBlockHeight,
+      };
+
+      let compactScan;
+      if (Object.values(flushReasons).some(Boolean)) {
+        compactScan = this.viewServer.flushUpdates();
+        console.log('flush', compactBlock, flushReasons, compactScan);
+
+        // in an atomic query, this
+        // - saves 'sctUpdates'
+        // - saves new decrypted notes
+        // - saves new decrypted swaps
+        // - updates last block synced
+        await this.indexedDb.saveScanResult(compactScan);
+
+        // - detect unknown asset types, query the chain for asset metadata
+        // - or, generate default fallback metadata
+        // - update idb
+        // this makes a network query.
+        // TODO: should this be scanning swaps as well?
+        await this.identifyNewAssets(compactScan.newNotes);
+      }
+
+      // nullifiers on this block may match notes or swaps from db
+      // mark them as spent/claimed, return what was used
+      const spentNullifiers = await this.resolveNullifiers(
+        compactBlock.nullifiers,
+        compactBlock.height,
+      );
+      // commitments on this block?? idk man
+      const commitmentRecords = await this.resolveCommitments(
+        compactScan?.sctUpdates.store_commitments,
+      );
+
+      const fullScanReasons = {
+        'spent nullifiers': spentNullifiers.size,
+        'updated commitments': commitmentRecords.size,
+        'new notes': compactScan?.newNotes.length,
+        'new swaps': compactScan?.newSwaps.length,
+      };
+
+      if (Object.values(fullScanReasons).some(Boolean)) {
+        console.log('full scan', compactBlock.height, fullScanReasons);
+
+        // this is a network query
+        const fullBlock = await this.querier.app.txsByHeight(compactBlock.height);
```

Comment:
I think it should not be `fullBlock`, this variable stores the list of transactions of the block. Perhaps `blockTxs`?

Reply:
ok